indexes/utxoindex/src/index.rs

```diff
@@ -14,62 +14,49 @@ use consensus_core::{api::DynConsensus, tx::ScriptPublicKeys, utxo::utxo_diff::U
 use hashes::Hash;
 use kaspa_core::trace;
 use kaspa_utils::arc::ArcExtensions;
-use parking_lot::RwLock;
 use std::sync::Arc;
 
 const RESYNC_CHUNK_SIZE: usize = 2048; //Increased from 1k (used in go-kaspad), for quicker resets, while still having a low memory footprint.
 
-/// UtxoIndex indexes [`CompactUtxoEntryCollections`] by [`ScriptPublicKey`], commits them to its owns tore, and emits changes.
-#[derive(Clone)]
+/// UtxoIndex indexes [`CompactUtxoEntryCollections`] by [`ScriptPublicKey`], commits them to its owns store, and emits changes.
+/// Note: The UtxoIndex struct by itself is not thread save, thread-safety is only guaranteed by using the dynamic types in `utxoindex::core::api` with the same RwLock applied to all used dyn traits.
 pub struct UtxoIndex {
     consensus: DynConsensus,
-    store: Arc<RwLock<Store>>,
+    store: Store,
 }
 
 impl UtxoIndex {
-    /// Creates a new [`UtxoIndex`] listening to the passed consensus, and consensus receiver.
+    /// Creates a new [`UtxoIndex`] listening to the passed consensus.
     pub fn new(consensus: DynConsensus, db: Arc<DB>) -> Self {
```

Comment:
Let's make `new` return a locked UtxoIndex, so it won't be possible to use it unsafely